### PR TITLE
Github::UserLink - migrate the data and make use of the stored usernames

### PR DESCRIPTION
**Pull request by MariagraziaAlastra**
This is blocking the dev pipeline redesign, so we need this done ASAP.
Assigning to @jbarrett for review, still WIP

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
@jbarrett I think it's ready for review!

File: sql/1.095/github_user.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+    CREATE TABLE github_user_link (
+        "users_id" bigint NOT NULL,
+        "data" text NOT NULL
+    );
+
+    ALTER TABLE users ADD COLUMN "github_user" text;
+COMMIT;
```
Comment (jbarrett) [inline sql/1.095/github_user.sql]: This table already exists, we're just wrapping the table def.

Comment (jbarrett) [inline sql/1.095/github_user.sql]: Just noticing the users `ALTER` - we need to remove the CREATE TABLE block or this will fail.

Comment (MariagraziaAlastra) [inline sql/1.095/github_user.sql]: @jbarrett Ah, right, for some reason I thought it only existed on wing and not in production :)
Removed now!

File: lib/DDGC/DB/Result/GitHub/UserLink.pm [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+use Moose;
+extends 'DDGC::DB::Base::Result';
+use DBIx::Class::Candy;
+use namespace::autoclean;
+
+table 'github_user_link';
+
+primary_column users_id       => {data_type => 'bigint', is_nullable => 0};
+column data                   => {data_type => 'text',   is_nullable => 0, serializer_class => 'JSON'};
```
Comment (jbarrett) [inline lib/DDGC/DB/Result/GitHub/UserLink.pm]: My bad - this column is called `gh_data`

Comment (MariagraziaAlastra) [inline lib/DDGC/DB/Result/GitHub/UserLink.pm]: @jbarrett Np, renamed!

**Comment by jbarrett** [comment]
Thanks @MariagraziaAlastra!

Appears to be working, we can probably go ahead with this today.

**Comment by MariagraziaAlastra** [comment]
Nice, thanks @jbarrett! I'll merge then :)

**Pull request merged**